core/arch/opnd_shared.c

```diff
@@ -2099,12 +2099,75 @@ reg_32_to_opsz(reg_id_t reg, opnd_size_t sz)
     return reg;
 }
 
+static reg_id_t
+reg_resize_to_zmm(reg_id_t simd_reg)
+{
+#ifdef X86
+    if (reg_is_strictly_xmm(simd_reg)) {
+        return simd_reg - DR_REG_START_XMM + DR_REG_START_ZMM;
+    } else if (reg_is_strictly_ymm(simd_reg)) {
+        return simd_reg - DR_REG_START_YMM + DR_REG_START_ZMM;
+    } else if (reg_is_strictly_zmm(simd_reg)) {
+        return simd_reg;
+    }
+    CLIENT_ASSERT(false, "Not a simd register.");
+#endif
+    return DR_REG_INVALID;
+}
+
+static reg_id_t
+reg_resize_to_ymm(reg_id_t simd_reg)
+{
+#ifdef X86
+    if (reg_is_strictly_xmm(simd_reg)) {
+        return simd_reg - DR_REG_START_XMM + DR_REG_START_YMM;
+    } else if (reg_is_strictly_ymm(simd_reg)) {
+        return simd_reg;
+    } else if (reg_is_strictly_zmm(simd_reg)) {
+        return simd_reg - DR_REG_START_ZMM + DR_REG_START_YMM;
+    }
+    CLIENT_ASSERT(false, "not a simd register.");
+#endif
+    return DR_REG_INVALID;
+}
+
+static reg_id_t
+reg_resize_to_xmm(reg_id_t simd_reg)
+{
+#ifdef X86
+    if (reg_is_strictly_xmm(simd_reg)) {
+        return simd_reg;
+    } else if (reg_is_strictly_ymm(simd_reg)) {
+        return simd_reg - DR_REG_START_YMM + DR_REG_START_XMM;
+    } else if (reg_is_strictly_zmm(simd_reg)) {
+        return simd_reg - DR_REG_START_ZMM + DR_REG_START_XMM;
+    }
+    CLIENT_ASSERT(false, "not a simd register");
+#endif
+    return DR_REG_INVALID;
+}
+
 reg_id_t
 reg_resize_to_opsz(reg_id_t reg, opnd_size_t sz)
```

Comment:
Please add the new description to the docs in opnd.h

Reply:
Done.